client/my-sites/customer-home/main.jsx

```diff
@@ -0,0 +1,258 @@
+import EmptyContent from 'components/empty-content';
+import ChecklistWpcom from 'my-sites/checklist/main';
+import JetpackChecklist from 'my-sites/plans/current-plan/jetpack-checklist';
+import QueryJetpackPlugins from 'components/data/query-jetpack-plugins';
+import Main from 'components/main';
+import VerticalNav from 'components/vertical-nav';
+import VerticalNavItem from 'components/vertical-nav/item';
+import { preventWidows } from 'lib/formatting';
+import SidebarNavigation from 'my-sites/sidebar-navigation';
+import { getSelectedSite, getSelectedSiteId, getSelectedSiteSlug } from 'state/ui/selectors';
+import {
+	isJetpackSite,
+	getCustomizerUrl,
+	canCurrentUserUseCustomerHome,
+} from 'state/sites/selectors';
+import isSiteAutomatedTransfer from 'state/selectors/is-site-automated-transfer';
+import PageViewTracker from 'lib/analytics/page-view-tracker';
+import DocumentHead from 'components/data/document-head';
+import getSiteChecklist from 'state/selectors/get-site-checklist';
+import QuerySiteChecklist from 'components/data/query-site-checklist';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+const ActionBox = ( { href, onClick, iconSrc, label } ) => {
+	const buttonAction = { href, onClick };
+	return (
+		<div className="customer-home__box-action">
+			<Button { ...buttonAction }>
+				<img src={ iconSrc } alt="" />
+				<span>{ label }</span>
+			</Button>
+		</div>
+	);
+};
+
+const isPendingTask = task => false === task.isCompleted;
+
+class Home extends Component {
+	static propTypes = {
+		checklistMode: PropTypes.string,
+
+		site: PropTypes.object.isRequired,
+		siteId: PropTypes.number.isRequired,
+		siteSlug: PropTypes.string.isRequired,
+		customizeUrl: PropTypes.string.isRequired,
+		canUserUseCustomerHome: PropTypes.bool.isRequired,
+		hasChecklistData: PropTypes.bool.isRequired,
+		shouldRenderChecklist: PropTypes.bool.isRequired,
+		isJetpackNotAtomic: PropTypes.bool.isRequired,
+	};
+
+	render() {
+		const { translate, canUserUseCustomerHome } = this.props;
+
+		if ( ! canUserUseCustomerHome ) {
+			return (
+				<EmptyContent
+					title={ preventWidows( translate( 'You are not authorized to view this page.' ) ) }
+					line={ preventWidows(
+						translate(
+							"Ensure you have selected the right site on the sidebar. If it's correct, you might want to contact the administrator of this site."
+						)
+					) }
+					illustration="/calypso/images/illustrations/error.svg"
+				/>
+			);
+		}
+
+		const { siteId, hasChecklistData, shouldRenderChecklist } = this.props;
+
+		return (
+			<Main className="customer-home__main is-wide-layout">
+				<PageViewTracker path={ `/home/:site` } title={ translate( 'Customer Home' ) } />
+				<DocumentHead title={ translate( 'Customer Home' ) } />
+				<SidebarNavigation />
+				{ siteId && ! hasChecklistData && <QuerySiteChecklist siteId={ siteId } /> }
+				{ shouldRenderChecklist ? this.renderChecklist() : this.renderCustomerHome() }
```

Comment:
I'm not sure if this is the best way to do it, but this uses `hasChecklistData` to stop the Customer  Home displaying while the data is fetched.

```suggestion
				{ shouldRenderChecklist && this.renderChecklist() }
				{ hasChecklistData && ! shouldRenderChecklist && this.renderCustomerHome() }
```